test/unit/ajax.js

```diff
@@ -837,6 +837,92 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - do not execute scripts from unsuccessful responses (gh-4250)", 7, function( assert ) {
+		var globalEval = jQuery.globalEval;
+
+		function request( title, options, overrides ) {
```

Comment:
Why do we need both `options` & `overrides`? Can't we drop `options` & rename `overrides` to `options`? Since `beforeSend` & a few others are overriding what's in `options`, there doesn't seem to be a reason to put those things in `options`. But if we don't put them there but just in `overrides` then changing:
```js
jQuery.extend( options, {...}, overrides )
```
to
```js
jQuery.extend( {...}, options, overrides )
```
wouldn't change anything and then we can just merge `options` with `overrides` instead of keeping two objects.